kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -488,6 +488,110 @@ class ResidualBasedBlockBuilderAndSolver
+     * @param MoveMesh tells if the update of the scheme needs  to be performed when calling the Update of the scheme
+     */
+    void BuildAndSolveLinearizedOnPreviousIteration(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart& rModelPart,
+        TSystemMatrixType& rA,
+        TSystemVectorType& rDx,
+        TSystemVectorType& rb,
+        const bool MoveMesh
+        ) override
+    {
+        KRATOS_INFO_IF("BlockBuilderAndSolver", this->GetEchoLevel() > 0)
+            << "Linearizing on Old iteration" << std::endl;
+
+        if(rModelPart.GetBufferSize() == 1) {
+            KRATOS_ERROR << "BlockBuilderAndSolver: \n"
+                << "The buffer size needs to be at least 2 in order to use \n"
+                << "BuildAndSolve_LinearizedOnOldIteration \n"
+                << "current buffer size for modelpart: " << rModelPart.Name() << std::endl
+                << "is :" << rModelPart.GetBufferSize()
+                << " Old Stiffness on First Iteration. \n"
+                << "setting mUseOldStiffnessInFirstIteration=false " << std::endl;
+        }
+
+        DofsArrayType FixedDofs;
+        for(auto& dof : BaseType::mDofSet){
+            if(dof.IsFixed()){
+                FixedDofs.push_back(&dof);
+                dof.FreeDof();
+            }
+        }
+
+        //TODO: Here we need to take the vector from other ones because
+        // We cannot create a trilinos vector without a communicator. To be improved!
+        TSystemVectorType dx_prediction(rDx);
+        TSystemVectorType rhs_addition(rb); //we know it is zero here, so we do not need to set it
+
+        // Here we bring back the database to before the prediction,
+        // but we store the prediction increment in dx_prediction.
+        // The goal is that the stiffness is computed with the
+        // converged configuration at the end of the previous step.
+        const auto it_dof_begin = BaseType::mDofSet.begin();
+        #pragma omp parallel for
+        for (int i = 0; i < static_cast<int>(BaseType::mDofSet.size()); ++i) {
+            auto it_dof = it_dof_begin + i;
+            //NOTE: this is initialzed to - the value of dx prediction
+            dx_prediction[it_dof->EquationId()] = -(it_dof->GetSolutionStepValue() - it_dof->GetSolutionStepValue(1));
+        }
+
+        // Use UpdateDatabase to bring back the solution to how it was at the end of the previous step
+        pScheme->Update(rModelPart, BaseType::mDofSet, rA, dx_prediction, rb);
+        const auto it_node_begin = rModelPart.NodesBegin();
+        if (MoveMesh) {
+            #pragma omp parallel for
+            for (int i = 0; i < static_cast<int>(rModelPart.NumberOfNodes()); ++i) {
+                auto it_node = it_node_begin + i;
+                noalias(it_node->Coordinates()) = it_node->GetInitialPosition().Coordinates()
+                                                + it_node->FastGetSolutionStepValue(DISPLACEMENT);

```

Comment:
hmmmmm
Hardcoding variables in the B&S ...?

Reply:
DISPLACEMENT is a special case (and it has always being treated as such)

we always assume that X = X0+DISPLACEMENT_X (the move mesh of the strategy always did that)